### PR TITLE
add logaddexp api

### PR DESCRIPTION
**Pull request by zhiboniu**
### PR types

New features
### PR changes

APIs
### Description

add logaddexp api

中文文档:
https://github.com/PaddlePaddle/docs/pull/5845

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_logaddexp.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import unittest
+
+import numpy as np
+
+import paddle
+
+
+def ref_logaddexp_old(x, y):
+    y = np.broadcast_to(y, x.shape)
+    out = np.log(1 + np.exp(np.minimum(x, y) - np.maximum(x, y))) + np.maximum(
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/test_logaddexp.py]: it is better to use `np.log1p` and `np.absolute`?

File: python/paddle/__init__.py [hunk 2]
```diff
@@ -258,6 +258,7 @@
 from .tensor.math import add  # noqa: F401
 from .tensor.math import subtract  # noqa: F401
 from .tensor.math import logsumexp  # noqa: F401
+from .tensor.math import logaddexp  # noqa: F401
```
Comment (jeff41404) [inline python/paddle/__init__.py:261]: add `logaddexp` in \_\_all\_\_ list below ?

File: python/paddle/fluid/tests/unittests/test_logaddexp.py (hunk 1 above)
Comment (zhiboniu) [reply python/paddle/fluid/tests/unittests/test_logaddexp.py]: done

File: python/paddle/__init__.py (hunk 2 above)
Comment (zhiboniu) [reply python/paddle/__init__.py:261]: done

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
LGTM

**Pull request merged**